commit 011fd22c89b924417c10ec5e47380b7f795f35cd
Author: Eric Anholt <eric@anholt.net>
Date:   Wed Jun 7 14:39:49 2017 -0700

    drm/vc4: Add FB modifier support to firmwarekms.
    
    Signed-off-by: Eric Anholt <eric@anholt.net>
    (cherry picked from commit 11752d73488e08aaeb65fe8289a9c016acde26c2)

diff --git a/drivers/gpu/drm/vc4/vc4_firmware_kms.c b/drivers/gpu/drm/vc4/vc4_firmware_kms.c
index 809437cf3215..d5066f395b62 100644
--- a/drivers/gpu/drm/vc4/vc4_firmware_kms.c
+++ b/drivers/gpu/drm/vc4/vc4_firmware_kms.c
@@ -17,6 +17,7 @@
 #include "drm/drm_atomic_helper.h"
 #include "drm/drm_plane_helper.h"
 #include "drm/drm_crtc_helper.h"
+#include "drm/drm_fourcc.h"
 #include "linux/clk.h"
 #include "linux/debugfs.h"
 #include "drm/drm_fb_cma_helper.h"
@@ -134,6 +135,10 @@ static void vc4_primary_plane_atomic_update(struct drm_plane *plane,
 	fbinfo->yoffset = state->crtc_y;
 	fbinfo->base = bo->paddr + fb->offsets[0];
 	fbinfo->pitch = fb->pitches[0];
+
+	if (fb->modifier == DRM_FORMAT_MOD_BROADCOM_VC4_T_TILED)
+		fbinfo->bpp |= BIT(31);
+
 	/* A bug in the firmware makes it so that if the fb->base is
 	 * set to nonzero, the configured pitch gets overwritten with
 	 * the previous pitch.  So, to get the configured pitch
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1127_011fd22c89b924417c10ec5e47380b7f795f35cd.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1127_011fd22c89b924417c10ec5e47380b7f795f35cd.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
